browser/ipfs/ipfs_tab_helper.cc

```diff
@@ -62,19 +111,115 @@ bool IPFSTabHelper::MaybeCreateForWebContents(
   return true;
 }
 
+void IPFSTabHelper::HostResolvedCallback(
+    const std::string& host,
+    const std::vector<std::string>& text_results) {
+  GURL current = web_contents()->GetURL();
+  if (current.host() != host || !current.SchemeIsHTTPOrHTTPS())
+    return;
+  std::string dnslink = GetDNSRecordValue(text_results, kDnsLinkHeader);
+  if (dnslink.empty())
+    return;
+  ipfs_resolved_host_ = host;
+  if (pref_service_->GetBoolean(kIPFSAutoRedirectDNSLink)) {
+    content::OpenURLParams params(GetIPFSResolvedURL(), content::Referrer(),
+                                  WindowOpenDisposition::CURRENT_TAB,
+                                  ui::PAGE_TRANSITION_LINK, false);
+    web_contents()->OpenURL(params);
```

Comment:
what happens if there is a txt record for both the hostname and _dnslink.hostname? It seems like once we get a valid result we should abort any other in-progress requests? Also since the hostname request will be cached it seems like it would be better to only run the _dnslink.hostname query if the hostname query doesn't have a dnslink record?

Reply:
we do not know what domain will have TXTrecord with dnslink, so we have to check both all time. Once we ignore dnslink value and should only check for presence of this TXT record it doenst matter for us which one will be finished first.